UPGRADE-1.12.md

```diff
@@ -84,6 +84,115 @@ with test or remove these services with complier pass.
 9. Passing a `Gaufrette\Filesystem` to `Sylius\Component\Core\Uploader\ImageUploader` constructor is deprecated since
 Sylius 1.12 and will be prohibited in 2.0. Use `Sylius\Component\Core\Filesystem\Adapter\FlysystemFilesystemAdapter` instead.
 
+### Api clients refactor
+
+#### New way of defining the clients
+
+In `1.12` version we changed the way of defining API clients in our behat contexts.
+We expanded the  `Sylius/Behat/Client/ApiClientInterface` methods with extra parameter `$resource`.
+
+  ```diff
+  -  public function index(): Response
+  +  public function index(string $resource): Response
+    
+  -  public function subResourceIndex(string $subResource, string $id): Response;
+  +  public function subResourceIndex(string $resource, string $subResource, string $id): Response;
+    
+  -  public function show(string $id): Response;
+  +  public function show(string $resource, string $id): Response;
+  
+  -  public function delete(string $id): Response;
+  +  public function delete(string $resource, string $id): Response;
+  
+  -  public function applyTransition(string $id, string $transition, array $content = []): Response;
+  +  public function applyTransition(string $resource, string $id, string $transition, array $content = []): Response;
+  
+  -  public function customItemAction(string $id, string $type, string $action): Response;
+  +  public function customItemAction(string $resource, string $id, string $type, string $action): Response;
+  
+  -  public function buildCreateRequest(): void;
+  +  public function buildCreateRequest(string $resource): void;
+  
+  -  public function buildUpdateRequest(string $id): void;
+  +  public function buildUpdateRequest(string $resource, string $id): void;
+  
+  -  public function buildUploadRequest(): void;
+  +  public function buildUploadRequest(string $resource): void;
+  ```
+
+With the follow change we are no more defining the clients in `api.xml` file (there are only defined the `shop` and `admin` endpoints) and now to make a call by a different client you have to pass it inside of the methods above.
+You can see the actual difference in example below:
+
+  ```diff
+  -  $this->avatarImagesClient->buildUploadRequest();
+  +  $this->client->buildUploadRequest(Resources::AVATAR_IMAGES);
+  ```
+
+As you can see the `Sylius\Behat\Context\Api\Resources` class has all the defined clients stored in constances.
+We also removed the `Sylius\Behat\ClientApiIriClient` alongside with `Sylius\Behat\ClientApiIriClientInterface` as it's no more used.
+
+#### Changes in contexts
+
+The constructors of the behat contexts changed with the improvement.
+Here is the example change of the constructor of `Sylius\Behat\Context\Api\Shop\OrderContexts`
```

Comment:
```suggestion
Here is the example change of the constructor of `Sylius\Behat\Context\Api\Shop\OrderContext`
```
Though IMO, it'd be better to show an example where the number of clients actually changed.